File: LinearSearch.java

public class LinearSearch {

    static int searchR(int a[], int num, int left, int right) {
	if (left > right) {
	    return -1;
	}
	if (num == a[right]) {
	    return right;
	}
	if (num == a[left]) {
	    return left;
	}
	return searchR(a, num, left+1, right-1);
    }

    public static void main(String[] args) {
	int a[] = {1, 2, 3, 4, 11, 6 };
	int num = 11;
	
	// using iterative approach
	System.out.println(searchR(a, num, 0, a.length-1));
    }

}

File: linear_search.c
#include <stdio.h>

int search_r(int a[], int index, int length, int num)
{
	if (index == -1)
		return -1;
	else if (a[index] == num)
		return index;
	else
		return search_r(a, index+1, length, num);
}

int main(void)
{
	int a[] = {2, 3, 44, 0, 7, 6};
	printf("%d\n", search_r(a, 0, 6, 7));
}
